packages/reactive-element/src/test/css-tag_test.ts

```diff
@@ -99,5 +100,60 @@ suite('Styling', () => {
       const bodyStyles = `${cssModule}`;
       assert.equal(bodyStyles.replace(/\s/g, ''), '.my-module{color:yellow;}');
     });
+
+    test('`adoptStyles uses different cached `styleSheets` per document', async () => {
+      if (supportsAdoptingStyleSheets) {
+        const div = document.createElement('div');
+        document.body.appendChild(div);
+        const shadowRoot = div.attachShadow({mode: 'open'});
+        const styles = [
+          css`
+            div {
+              margin: 2px solid red;
+            }
+          `,
+          css`
+            div {
+              padding: 10px;
+            }
+          `,
+        ];
+        adoptStyles(shadowRoot, styles);
+        const adoptedStyleSheets_1 = [...shadowRoot.adoptedStyleSheets];
+        adoptStyles(shadowRoot, styles);
+        const adoptedStyleSheets_2 = [...shadowRoot.adoptedStyleSheets];
+        adoptedStyleSheets_1.forEach((_, index) => {
+          assert.strictEqual(
+            adoptedStyleSheets_1[index],
+            adoptedStyleSheets_2[index]
+          );
+        });
+        const iframe = document.createElement('iframe');
+        const html = '<body>Foo</body>';
+        document.body.appendChild(iframe);
+        iframe.contentWindow?.document.open();
```

Comment:
I don't think you need to do this. I think `iframe.contentDocument.body` should just exist.

Reply:
Oh I see. Affirmative.